### ISSUE TITLE
Android 13 is not supported

### ISSUE THREAD

**Issue by darrenmeehan** [publifactory/Overland_android#39]
I was hoping to install this but the Play Store says my phone isn’t supported.

With Android 14 coming out soon more and more phones will be on these versions.

**Comment by samwilson** [comment]
I get the impression that this project isn't very active.

**Comment by gedankenstuecke** [comment]
I fear one big problem that we are currently running into with this is that the deployment to the Play Store was/is handled by the former "Centre for Research & Interdisciplinarity", where none of us have connects to any longer due to a change in focus on their side, making it hard for us to even deploy any updates at the moment!

**Comment by darrenmeehan** [comment]
Is it something they're open to providing access to?

**Comment by gedankenstuecke** [comment]
I'd not be too sure tbh, as they are highly unresponsive. And I fear I'm not an Android person myself, so I'd not be best poised to update & get an updated deployed version into the Play store!

**Comment by darrenmeehan** [comment]
Thanks for the insight, is there anything ye could do with help on? Could this be published under a new account?

**Comment by gedankenstuecke** [comment]
Absolutely it could be, it's a matter of time for shuffling it through the review process, if someone wanted to invest that time and deploy it, then I'd be happy to support as much as I can !

**Comment by jeverling** [comment]
Maybe it would be possible to provide an apk to be installed manually?

**Comment by darrenmeehan** [comment]
I'm happy to help. Initially having an APK to install would unblock the project I want to work on.

I think having some CI like GitHub Actions would be an ideal means of building and deploying to the play store. I made a start on this here https://github.com/darrenmeehan/Overland_android/actions/runs/7348202268/job/20005897654

This thread has grown some legs, there's several issues that need addressed here that I think warrant their own GitHub issue.

**Comment by gedankenstuecke** [comment]
That's great @darrenmeehan, having the APK for folks to install would help a lot, I'm not sure how easy it would be to automate the push to the play store (I've only gone through the iOS equivalent processes myself, which is highly manual).

**Comment by darrenmeehan** [comment]
From what I've seen online it should be possible but I think getting the APK building reliably is a better goal.

I'm looking into getting a Github dev container created so it's easier to collaborate.

The trouble I'm having at the moment is knowing what versions of tooling to install. I'm by no means an android developer! Once that's sorted the Github Action will be a lot easier.

**Comment by darrenmeehan** [comment]
@gedankenstuecke are there any docs on what version of Java, Gradle and other Android related SDKs the app can be built with?

From what I can see the APK hasn't been built since 2021.

**Comment by gedankenstuecke** [comment]
I fear I don't know! @nicolaseberle would you happen to remember by chance or would be able to check? :)

**Comment by darrenmeehan** [comment]
I'll get the Codespaces setup I have up for review so we can talk through.

I was hitting several issues mostly related to some react native npm packages.

My development environment is a bit all over the place as I'm travelling at the moniabd only have access to an iPad!

**Comment by darrenmeehan** [comment]
I have a draft PR up here https://github.com/OpenHumans/Overland_android/pull/40

**Comment by gedankenstuecke** [comment]
Perfect, thanks so much @darrenmeehan! I double checked on my end, and I fear with moving between computers (and jobs and countries!), I indeed don't have a local dev environment set up, otherwise I could have checked for the build settings I used to have! I hope that @nicolaseberle might be able to help once the holiday season is over! :)

**Comment by justdick** [comment]
Please any updates on this supporting newer version of android, or getting an apk to install?

**Comment by aaronpk** [comment]
I'd be happy to publish this in my Google developer account. I have no recent Android dev experience, only iOS for the Overland iOS app, but I do have a developer account I could use for this.

**Comment by gedankenstuecke** [comment]
That's very kind @aaronpk, we'd be super happy to have it published under your account! I think the main roadblock at this stage would be finding anyone capable of the Android dev enough to bump the versions to the minimum requirements for newer Android versions. As unfortunately, that person isn't me!

If anyone would like to take that on, I'd be happy to support getting it all merged in!

In the meanwhile, https://owntracks.org/ might be the closest alternative for Android, as Overland on iOS already supports the same dialect if I'm not mistaken?

**Comment by mbechtel** [comment]
I have been using Overland to send data to my personal tracking app for a few years now.  It has worked really well with minimal battery usage, and without issue the entire time.  I recently had to purchase a new phone and noticed that it is no longer available from Google Play.  I am not an Android developer, but my company does maintain an Android app and like @aaronpk, I would be happy to publish the app to my company's developer account if that will help.  I will try OwnTracks in the meantime, so thank you for that tip.

**Comment by dshanske** [comment]
@mbechtel I would come back to using this app if someone was keeping it up. I opted to switch back to GPSLogger for integrating with Compass.

**Comment by gedankenstuecke** [comment]
Yeah, the problem remains that we don't have someone that's an Android dev to do the necessary updates at the moment. If you know any Android devs that would like a free (software) puppie, we'd be happy to hand this over :see_no_evil:
